ssf: remove the reset_evt sdfw_update and echo service

hakonfam opened the pull request:
These are not needed

Ref: NCSDK-30980

NordicBuilder commented:
## CI Information

<sup>_To view the history of this post, clich the 'edited' button above_</sup>
Build number: 1
### Inputs:
#### Sources:
sdk-nrf: PR head: [7204ba48f9ddde9c1fcccf7129f925ee10248bac](https://github.com/nrfconnect/sdk-nrf/pull/19490/commits/7204ba48f9ddde9c1fcccf7129f925ee10248bac)
<details><summary> more details</summary>

sdk-nrf:
> PR head: [7204ba48f9ddde9c1fcccf7129f925ee10248bac](https://github.com/nrfconnect/sdk-nrf/pull/19490/commits/7204ba48f9ddde9c1fcccf7129f925ee10248bac)
> merge base: [e28af5e0cff6bde826d37613685c8c30ca1905d3](https://github.com/nrfconnect/sdk-nrf/commit/e28af5e0cff6bde826d37613685c8c30ca1905d3)
> target head (main): [c03a25425898bee90f44bd120d443ebe3def0a58](https://github.com/nrfconnect/sdk-nrf/commit/c03a25425898bee90f44bd120d443ebe3def0a58)
> [Diff](https://github.com/nrfconnect/sdk-nrf/compare/e28af5e0cff6bde826d37613685c8c30ca1905d3..7204ba48f9ddde9c1fcccf7129f925ee10248bac)</details>

#### Github labels
| Enabled | Name | Description |
| -- | -- | -- |
|| ci-disabled | _Disable the ci execution_ | 
|| ci-all-test | _Run all of ci, no test spec filtering will be done_ | 
|| ci-force-downstream | _Force execution of downstream even if twister fails_ | 
|| ci-run-twister | _Force run twister_ | 
|| ci-run-zephyr-twister | _Force run zephyr twister_ | 

<details><summary> List of changed files detected by CI (27)</summary>

```
include
│  ├── sdfw
│  │  ├── sdfw_services
│  │  │  ├── echo_service.h
│  │  │  │ reset_evt_service.h
samples
│  ├── sdfw
│  │  ├── ssf_client
│  │  │  ├── Kconfig
│  │  │  ├── src
│  │  │  │  │ main.c
│  ├── suit
│  │  ├── recovery
│  │  │  │ prj.conf
subsys
│  ├── sdfw_services
│  │  ├── services
│  │  │  ├── CMakeLists.txt
│  │  │  ├── Kconfig
│  │  │  ├── echo
│  │  │  │  ├── CMakeLists.txt
│  │  │  │  ├── Kconfig
│  │  │  │  ├── echo_service.c
│  │  │  │  ├── echo_service.cddl
│  │  │  │  ├── zcbor_generated
│  │  │  │  │  ├── CMakeLists.txt
│  │  │  │  │  ├── echo_service_decode.c
│  │  │  │  │  ├── echo_service_decode.h
│  │  │  │  │  ├── echo_service_encode.c
│  │  │  │  │  ├── echo_service_encode.h
│  │  │  │  │  │ echo_service_types.h
│  │  │  ├── reset_evt
│  │  │  │  ├── CMakeLists.txt
│  │  │  │  ├── Kconfig
│  │  │  │  ├── reset_evt_service.c
│  │  │  │  ├── reset_evt_service.cddl
│  │  │  │  ├── zcbor_generated
│  │  │  │  │  ├── CMakeLists.txt
│  │  │  │  │  ├── reset_evt_service_decode.c
│  │  │  │  │  ├── reset_evt_service_decode.h
│  │  │  │  │  ├── reset_evt_service_encode.c
│  │  │  │  │  ├── reset_evt_service_encode.h
│  │  │  │  │  │ reset_evt_service_types.h
```
</details>

### Outputs:

#### Toolchain
Version: 
Build docker image: 

#### Test Spec & Results: :white_check_mark: Success; :x: Failure; :orange_circle: Queued; :yellow_circle: Progress; :white_medium_square: Skipped; :warning: Quarantine
- :orange_circle: Toolchain
- :orange_circle: Build twister
- :orange_circle: Integration tests
   - :orange_circle: test-sdk-dfu
   - :orange_circle: test-secdom-samples-public
   - :warning: test-sdk-dfu

<details><summary>Disabled integration tests</summary>

- 
   - desktop52_verification
   - doc-internal
   - test_ble_nrf_config
   - test-fw-nrfconnect-apps
   - test-fw-nrfconnect-ble_mesh
   - test-fw-nrfconnect-ble_samples
   - test-fw-nrfconnect-boot
   - test-fw-nrfconnect-chip
   - test-fw-nrfconnect-fem
   - test-fw-nrfconnect-nfc
   - test-fw-nrfconnect-nrf-iot_libmodem-nrf
   - test-fw-nrfconnect-nrf-iot_lwm2m
   - test-fw-nrfconnect-nrf-iot_mosh
   - test-fw-nrfconnect-nrf-iot_nrf_provisioning
   - test-fw-nrfconnect-nrf-iot_positioning
   - test-fw-nrfconnect-nrf-iot_samples
   - test-fw-nrfconnect-nrf-iot_serial_lte_modem
   - test-fw-nrfconnect-nrf-iot_thingy91
   - test-fw-nrfconnect-nrf-iot_zephyr_lwm2m
   - test-fw-nrfconnect-nrf_crypto
   - test-fw-nrfconnect-ps
   - test-fw-nrfconnect-rpc
   - test-fw-nrfconnect-rs
   - test-fw-nrfconnect-tfm
   - test-fw-nrfconnect-thread
   - test-fw-nrfconnect-zigbee
   - test-low-level
   - test-sdk-audio
   - test-sdk-find-my
   - test-sdk-mcuboot
   - test-sdk-pmic-samples
   - test-sdk-sidewalk
   - test-sdk-wifi
</details>



*Note: This message is automatically posted and updated by the CI*

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='https://ncsdoc.z6.web.core.windows.net/PR-19490
/'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publish GitHub Action.*

jonathannilsen commented on samples/sdfw/ssf_client/src/main.c:
```diff
@@ -74,14 +38,6 @@ int main(void)
-
 	if (IS_ENABLED(CONFIG_ENABLE_SDFW_UPDATE_REQUEST)) {
 		sdfw_update();
 	}
```
Seems a bit strange to keep the sample here when this is disabled by default and so the sample effectively does nothing. Should we remove the sample entirely? Users should not be using this API for updating SDFW anyway.

anhmolt replied:
I agree. If the sample is not changed to do something else with sdfw_services, then it is of no use after removing these services.

jonathannilsen replied:
The only immediate use for this I can see would be for #19278. I still think we can remove `sdfw_update` here though, and even the whole `sdfw_update` client

anhmolt replied:
Agreed. Another possible suggestion is to invoke `ssf_psa_generate_random` and get some random values. Given that it is that simple, and no PSA initialization is required.

anhmolt reviewed (approved):
Listed files `echo_service.h` and `reset_evt_service.h` in `nrf/.github/test-spec.yml` can be removed. Also, are some of the other sdfw services missing from the `CI-secdom-test` list?

hakonfam commented:
> Listed files `echo_service.h` and `reset_evt_service.h` in `nrf/.github/test-spec.yml` can be removed. Also, are some of the other sdfw services missing from the `CI-secdom-test` list?

I removed these entries.

jonathannilsen reviewed (commented):
Looks like there are some warnings in the codeowners file now, probably need to remove some entries there
https://github.com/nrfconnect/sdk-nrf/actions/runs/12352771769/job/34470498305?pr=19490

jonathannilsen commented on subsys/sdfw_services/services/Kconfig:
```diff
@@ -5,10 +5,8 @@
 rsource "extmem/Kconfig"
 rsource "psa_crypto/Kconfig"
-rsource "reset_evt/Kconfig"
 rsource "sdfw_update/Kconfig"
```
```suggestion
```
This is causing build failures now

anhmolt reviewed (approved):
LGTM

One last ting, entry `"samples/sdfw/ssf_client/**/*"` should be removed from `test_spec.yml`. Now that the sample is removed.

hakonfam commented:
CI is green on rerun

umapraseeda commented:
Missing changelog entry

hakonfam commented:
> Missing changelog entry

Added an entry, please check if I added it to the right place - long time since I did an update to the sdk-nrf changelog.

peknis commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -354,7 +354,7 @@ Protocol serialization samples
 ------------
 
-|no_changes_yet_note|
+* Removed the ``ssf_client`` sample since all services being shown has been removed.
```
```suggestion
* Removed the SDFW: Service Framework Client sample since all related services are removed.
```

peknis replied:
Not so sure about the sample removal just like that. Shouldn't it first be stated as deprecated and removed only in later release?

hakonfam replied:
The server side implementation will be removed, so it has to be removed.

peknis replied:
Where is that then? Was it declared deprecated before removing it?

peknis replied:
@hakonfam You haven't still updated the text according to my suggestion.

peknis commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -476,7 +476,7 @@ nRF RPC libraries
 ---------------
 
-|no_changes_yet_note|
+* Removed the unused SDFW services ``echo_service``, ``reset_evt_service`` and ``sdfw_update_service``.
```
```suggestion
* Removed the unused SDFW services ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.
```

peknis replied:
This entry is a bit suspicious, as we don't seem to mention these services in any of the NCS libraries.

hakonfam replied:
They are sample services that are not used by any other library.

peknis replied:
I'd say we can drop this entry.

hakonfam commented:
@ncs-test-leads and @nrfconnect/ncs-aurora-doc please review - green CI

conflicting files are deleted

jonathannilsen commented:
> @ncs-test-leads and @nrfconnect/ncs-aurora-doc please review - green CI
> 
> conflicting files are deleted

* @nrfconnect/ncs-test-leads

FrancescoSer commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -536,7 +536,7 @@ nRF RPC libraries
 ---------------
 
-|no_changes_yet_note|
+* Removed the unused SDFW services ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.
```
```suggestion
* Removed the following unused SDFW services: ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.
```

FrancescoSer commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -388,7 +388,7 @@ Protocol serialization samples
 ------------
 
-|no_changes_yet_note|
+* Removed the ``ssf_client`` sample since all services being shown has been removed.
```
```suggestion
* Removed the ``ssf_client`` sample, as all services demonstrated by the sample have been removed.
```

peknis replied:
I also have a comment regarding this entry. I thought it was implemented already, but no. ssf_client need to be opened and written as SDFW: Service Framework Client

hakonfam replied:
I joined your suggestions

FrancescoSer commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -536,7 +536,7 @@ nRF RPC libraries
 ---------------
 
-|no_changes_yet_note|
+* Removed the following unused SDFW services: ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.
```
Shouldn't this entry be moved in this page? https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/releases_and_maturity/abi_compatibility.html

hakonfam replied:
It will appear there as well when the SDFW version that removes them is released. The comment in this changelog is referring to the removal of the client side implementation of the services.
